packages/client/machines/usersUniversalSearcMachine.ts

```diff
@@ -0,0 +1,311 @@
+import {
+    PaginatedUserSummariesSearchResult,
+    UserSummary,
+} from '@musicroom/types';
+import { EventFrom, Sender } from 'xstate';
+import { createModel } from 'xstate/lib/model';
+import { getFakeUserID } from '../contexts/SocketContext';
+import {
+    fetchMyFollowers,
+    fetchUserFollowers,
+    fetchUserFollowing,
+} from '../services/UsersSearchService';
+import { appScreenHeaderWithSearchBarMachine } from './appScreenHeaderWithSearchBarMachine';
+
+//TODO USE SEARCH MACHINE
+const usersUniversalSearchModel = createModel(
+    {
+        usersSummaries: [] as UserSummary[],
+        hasMore: true,
+        nextPage: 1,
+        searchQuery: '',
+    },
+    {
+        events: {
+            SUBMITTED: (searchQuery: string) => ({ searchQuery }),
+            CLEAR_QUERY: () => ({}),
+            CANCEL: () => ({}),
+
+            FETCHED_USERS: (args: {
+                usersSummaries: UserSummary[];
+                hasMore: boolean;
+                page: number;
+            }) => args,
+
+            LOAD_MORE_ITEMS: () => ({}),
+            REFRESH: () => ({}),
+
+            FAILED_FETCHING_USERS: () => ({}),
+        },
+    },
+);
+
+const assignSearchQueryToContext = usersUniversalSearchModel.assign(
+    {
+        searchQuery: (_, event) => event.searchQuery,
+        nextPage: 1,
```

Comment:
Why would we ?
Unless I'm mistaken, setting the nextPage to 1 is enough. It will make the machine fetch the selected page, the server result will mutate the `hasMore` boolean depending on it's real value

Reply:
If the response comes one day, indeed, `hasMore` will be set to a meaningful value.
If the fetching fails or is hanging, `hasMore` will be in an undetermined state. When we initialize the machine, `hasMore` has a value of `true`. Here what we really do is resetting the context of the machine, so I think the `hasMore` should be reset too, to come back to a determined state.

I don't think it's worth a change. We just need to know that there is an unknown case here. This is not the first place where we do not reset `hasMore`, so the problem is more global. No need to change for me.

Reply:
True facts !
Should create a more global issue ?

Reply:
I don't think. I think it's okay to go on. This is not the most critical part of the application 😄.